src/history/HistoryTests.cpp

```diff
@@ -96,6 +96,105 @@ class TmpDirConfigurator : public Configurator
     }
 };
 
+struct CatchupMetrics
```

Comment:
This is the type of stuff I was talking about. Nice.

At the same time, maybe we want to break this up so that it maps to Work classes? ie, if I have a CatchupWork that spawns 2 types of Work, I don't need to flatten out the metrics in one structure but instead have metrics for each Work implementation.